code/drasil-lang/lib/Language/Drasil/Space.hs

```diff
@@ -43,6 +44,30 @@ data Space =
   | Actor String 
   | Function (NE.NonEmpty Primitive) Primitive
   | Void
+  -- | Clifford algebra objects (Clifs)
+  | Clif ClifShape
+  deriving (Eq, Show)
+
+-- | The dimension of a clif
+data Dimension where
+  -- | Fixed dimension
+  Fixed :: Int -> Dimension
+  -- | Variable dimension
+  VDim  :: String -> Dimension
+  deriving (Eq, Show)
+
+-- The shape of a clif object
+data ClifShape where
```

Comment:
Let's have a smaller `Shape` type without as much redundancy. There will be ample time to have more shapes of things (especially when it comes to representations), not point making this complicated.